pkg/engine/virtualmachinescalesets.go

```diff
@@ -718,3 +720,9 @@ func getVMSSDataDisks(profile *api.AgentPoolProfile) *[]compute.VirtualMachineSc
 	}
 	return &dataDisks
 }
+
+func addCustomTagsToVMScaleSets(cs *api.ContainerService, vm *compute.VirtualMachineScaleSet) {
+	for key, value := range cs.Properties.MasterProfile.CustomVMTags {
+		vm.Tags[key] = &value
```

Comment:
can we add a check to only add this tag if it doesn't already exist? I think we want to avoid allowing users to override the  static tag such as `aksEngineVersion`

Reply:
```
if v, ok := vm.Tags[key]; !ok  {
    vm.Tags[key] = &value
}
```

Reply:
same comment for addCustomTagsToVM

Reply:
Sure! Makes total sense not allowing someone to override a tag that already exists! Thanks for the comments!

Reply:
Done!